python/ray/data/impl/progress_bar.py

```diff
@@ -54,18 +59,29 @@ def __init__(self, name: str, total: int, position: int = 0):
             self._bar = None
 
     def block_until_complete(self, remaining: List[ObjectRef]) -> None:
+        t = threading.current_thread()
         while remaining:
-            done, remaining = ray.wait(remaining, fetch_local=False)
+            done, remaining = ray.wait(remaining, fetch_local=False, timeout=0.1)
             self.update(len(done))
```

Comment:
Note: This is assuming that tqdm will only trigger a display refresh if the count has changed, which appears to be the case after looking at the [source](https://github.com/tqdm/tqdm/blob/f3fb54eb161a9f5de13352e16a70a9960946605b/tqdm/std.py#L1246).